server/channels/app/notification_push.go

```diff
@@ -619,14 +619,18 @@ func DoesNotifyPropsAllowPushNotification(user *model.User, channelNotifyProps m
 	return ""
 }
 
-func DoesStatusAllowPushNotification(userNotifyProps model.StringMap, status *model.Status, channelID string) model.NotificationReason {
+func DoesStatusAllowPushNotification(userNotifyProps model.StringMap, status *model.Status, channelID string, isCRT bool) model.NotificationReason {
 	// If User status is DND or OOO return false right away
 	if status.Status == model.StatusDnd || status.Status == model.StatusOutOfOffice {
 		return model.NotificationReasonUserStatus
 	}
 
 	pushStatus, ok := userNotifyProps[model.PushStatusNotifyProp]
-	if (pushStatus == model.StatusOnline || !ok) && (status.ActiveChannel != channelID || model.GetMillis()-status.LastActivityAt > model.StatusChannelTimeout) {
+	sendOnlineNotification := status.ActiveChannel != channelID || //We are in a different channel
+		model.GetMillis()-status.LastActivityAt > model.StatusChannelTimeout || //It has been a while since we were last active on this channel
+		isCRT //Is CRT, so being active in a channel doesn't mean you are seeing thread activity
```

Comment:
This change does mean that users will now get push notifications for push notifications for threads that they have open if they're not actively interacting with the app. I'm trying to think if that'd be undesirable, but since we're wanting to lean towards being more noisy about notifications to avoid missed ones, this is probably okay

Reply:
Thanks for the details of those two cases. That makes sense to me then since case 1 definitely seems like it's likely to happen fairly often.

As for the "active thread", we sort of have that from the web socket side changes that @agnivade did to avoid broadcasting as many "user is typing" events, but it's separate from `status.ActiveChannel`. We should really look at unifying those at some point since I can't see any reason that they need to be handled differently